commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -48,6 +52,63 @@ function(generate_build_stats_wrappers)
 endfunction()
 
 
+# Generate the build stats compiler wrapper for a given CMake variable.
+#
+# the only difference between this and lang, is setting the proper
+# Cmake variable name, e.g., CMAKE_LANG_COMPILER  vs CMAKE_OP
+# we can resolve this by taking in name of the variable to set
+#
+# The intent of this function is pass in arbitrary cmake variables that
+# map to commands and generate suitable wrappers.
+#
+# Supported functions are C, CXX, Fortran, AR, LD, and RANLIB
+#
+#  TODO: wrap MPIEXEC!
+function(generate_build_stats_wrapper_for_op op_name variable_to_set)
+
+  # 'op_name' is the short name, like C, CXX, Fortran, LD, AR, RANLIB
+  # this is will give us a lowercase 
+  string(TOLOWER "${op_name}" op_lc)
+  set(op_wrapper
+    "${${PROJECT_NAME}_BINARY_DIR}/build_stat_${op_lc}_wrapper.sh")
+
+  # there's an issue here - if CMAKE_FOO is unset (whatever `variable_to_set` is)
+  # we need a to know the command - but CMake hasn't chosen one yet...
+  if ("${${variable_to_set}}" STREQUAL "")
+    message("-- " "${variable_to_set} is not set, but a wrapper has been requested. Asking CMake to find ${op_lc}")
+    find_program(${variable_to_set} "${op_lc}")
```

Comment:
Yes - that's good.

If we are going to be robust, then if `ALLOW_FIND` is `false` and those aren't defined, we should skip wrapping them. Set `CMAKE_X_COMPILER_ORIG` to `false` or `NOT-FOUND` (however you do that). Then in the install script only `install` if `foo_ORIG` is defined. 

I'll need to read about that:
https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html

Reply:
I think I've got this.

To test, I set LD without `ALLOW_FIND`.  This would mimic the C/CXX/Fortran compiler not being set.

It correctly determines CMAKE_LD is undefined and `ALLOW_FIND = FALSE`
```
-- BASE_BUILD_DIR_FOR_PYTHON='/ssd_scratch/src/Trilinos-jje/fix-build-stats3'
-- Generating build stats wrapper for C
-- Generating build stats wrapper for CXX
-- Not wrapping LD because CMAKE_LD=`CMAKE_LD` is not set
-- CMAKE_AR is not set, but a wrapper has been requested. Asking CMake to find ar
-- CMAKE_AR='/usr/bin/ar'
-- Generating build stats wrapper for AR
-- CMAKE_RANLIB is not set, but a wrapper has been requested. Asking CMake to find ranlib
-- CMAKE_RANLIB='/usr/bin/ranlib'
-- Generating build stats wrapper for RANLIB
```